File: src/typescript/readme_reader.test.ts
import { readReadme } from "./readme_reader";
import { promises as fs } from "fs";
import { join } from "path";

test("readReadme should return file content", async () => {
    const tempPath = join(__dirname, "temp_README.md");
    await fs.writeFile(tempPath, "Sample README content");

    const content = await readReadme(tempPath);
    expect(content).toBe("Sample README content");

    await fs.unlink(tempPath);
});

test("readReadme should throw error if file not found", async () => {
    await expect(readReadme("non_existent.md")).rejects.toThrow("README.md not found.");
});

File: src/java/src/test/java/com/example/ReadmeReaderTest.java
package com.example;

import org.junit.jupiter.api.Test;
import java.nio.file.*;
import static org.junit.jupiter.api.Assertions.*;

public class ReadmeReaderTest {

    @Test
    public void testReadReadme() throws Exception {
        Path tempFile = Files.createTempFile("README", ".md");
        Files.writeString(tempFile, "Sample README content");

        assertEquals("Sample README content", ReadmeReader.readReadme(tempFile.toString()));

        Files.delete(tempFile);
    }

    @Test
    public void testReadmeNotFound() {
        Exception exception = assertThrows(Exception.class, () -> {
            ReadmeReader.readReadme("non_existent.md");
        });

        assertEquals("README.md not found.", exception.getMessage());
    }
}
